### DISCUSSION TITLE
PostgreSQL: The Primary Key of an Array Field is of Type String Instead of Number

### DISCUSSION THREAD

**Discussion by tobiasiv** [payloadcms/payload#7432] [Feature Requests & Ideas]
### Payload Version

3.0.0-beta.71

### Node Version

v20.9.0

### Next.js Version

15.0.0-canary.77

### Describe the Bug

In PostgreSQL, the primary key of an array field is currently a string. For consistency, it should be a serial number instead."

### Reproduction Steps

1. Clone the [payload 3.0 demo](https://github.com/payloadcms/payload-3.0-demo)
2. Use PostgreSQL as the database
3. Add an array field to the pages collection:

```js
    {
      slug: 'pages',
      admin: {
        useAsTitle: 'title',
      },
      fields: [
        {
          name: 'title',
          type: 'text',
        },
        {
          name: 'content',
          type: 'richText',
        },
        {
          name: "array", 
          type: "array", 
          fields: [
            {
              name: "arrayField", 
              type: "text"
            }
          ]
        }
      ],
    }
```

4. Check the database and payload for the primary key type:

```js
  array?:
    | {
        arrayField?: string | null;
        id?: string | null;
      }[]
    | null;
```

**Comment by jmikrut** [comment]
Hey @tobiasiv — array and blocks actually do need a `uuid` (string-based) and can't use serial IDs due to a variety of reasons:

- Those IDs are used within the admin UI for drag and drop capabilities, and the ID needs to be unique across the entire page
- If versions are enabled, we can't leverage serial IDs because there could be many versions of a given array row - so we need a proper `uuid` rather than a serial ID there as well

We might be able to work around these items, but it worked out best to just allow array and block rows to have string-based uuids at this point. I will convert this to a feature request though so we can continue to track interest. If there is enough, we could evaluate what it would take to switch these IDs over to integer.
